### PR TITLE
make proxy custom resource

### PR DESCRIPTION
**Pull request by yuval-k**
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/3663

### CODE REVIEW

File: projects/gloo/pkg/api/compress/compress.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+func MarshalSpec(in resources.Resource) (v1.Spec, error) {
+
+	data, err := protoutils.MarshalMap(in)
+	if err != nil {
+		return nil, crd.MarshalErr(err, "resource to map")
+	}
+
+	delete(data, "metadata")
+	delete(data, "status")
```
Comment (kdorosh) [inline projects/gloo/pkg/api/compress/compress.go:129]: instead would it be better to delete all keys that aren't spec?

File: projects/gloo/pkg/api/compress/compress_test.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+				},
+				Listeners: []*v1.Listener{{BindAddress: "1234"}},
+			}
+
+			spec, err := MarshalSpec(p)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(spec).To(HaveKey("compressedSpec"))
+		})
+
+		It("should uncompress to the same thing", func() {
```
Comment (kdorosh) [inline projects/gloo/pkg/api/compress/compress_test.go:40]: do you think its worth asserting that the compressed version is in fact smaller?

File: projects/gloo/pkg/api/compress/compress.go (hunk 1 above)
Comment (yuval-k) [reply projects/gloo/pkg/api/compress/compress.go:129]: our protos don't have a spec field.
essentially the spec is everything that's not metdata or status

Comment (yuval-k) [reply projects/gloo/pkg/api/compress/compress.go:129]: see here:
https://github.com/solo-io/solo-kit/blob/f300a3f30ddf5fdcb1515b387359d53643c2bb4a/pkg/api/v1/clients/kube/crd/crd.go#L111

**Review by kdorosh — APPROVED** [review]
lgtm

**Pull request merged**